core/src/main/java/io/kestra/core/models/conditions/types/PublicHolidayCondition.java

```diff
@@ -0,0 +1,73 @@
+package io.kestra.core.models.conditions.types;
+
+import de.focus_shift.jollyday.core.HolidayManager;
+import de.focus_shift.jollyday.core.ManagerParameters;
+import io.kestra.core.exceptions.InternalException;
+import io.kestra.core.models.annotations.Example;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.conditions.Condition;
+import io.kestra.core.models.conditions.ConditionContext;
+import io.kestra.core.models.conditions.ScheduleCondition;
+import io.kestra.core.utils.DateUtils;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+
+import javax.validation.constraints.NotEmpty;
+import java.time.LocalDate;
+import java.util.Locale;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Condition to allow events on public holidays"
+)
+@Plugin(
+    examples = {
+        @Example(
+            full = true,
+            code = {
+                """
+                - conditions:
+                    - type: io.kestra.core.models.conditions.types.PublicHolidayCondition
+                      country: FR
+                """
+            }
+        )
+    }
+)
+public class PublicHolidayCondition extends Condition implements ScheduleCondition {
+    @NotEmpty
+    @Schema(
+        title = "The date to test",
+        description = "Can be any variable or any valid ISO 8601 datetime, default will use the trigger date"
+    )
+    @Builder.Default
+    @PluginProperty(dynamic = true)
+    private String date = "{{ trigger.date }}";
```

Comment:
Following Slack discussion, I'm ok with that but it should be consistent across all conditions so please do the change accordingly or stay with now()

Reply:
It has been decided to align all date based conditions to the trigger date